ql/src/java/org/apache/hadoop/hive/ql/parse/DDLSemanticAnalyzer.java

```diff
@@ -2061,11 +2057,9 @@ private void analyzeAlterTablePartMergeFiles(ASTNode ast,
       }
 
       // throw a HiveException for other than rcfile and orcfile.
-      if (!((inputFormatClass.equals(RCFileInputFormat.class) ||
-          (inputFormatClass.equals(OrcInputFormat.class))))) {
+      if (!(inputFormatClass.equals(RCFileInputFormat.class) || inputFormatClass.equals(OrcInputFormat.class))) {
```

Comment:
note: I think that instead of class.equals; we should use marker interfaces

Reply:
indeed, that should be fixed at a later change.